### PR TITLE
docs: Remove mention of OWNERS files from contrib guide

### LINKED ISSUE #30: Remove mention of owner files in contributing guide
For Kata Containers we have dispensed with the `OWNERS` files we used to use in Clear Containers. The reason being it is pointless having to maintain these files along with the `.pullapprove.yml` files which encode the same information (and more).

Remove mention of the `OWNERS` files which was an oversight when migrating the doc.

### PR DESCRIPTION
**Pull request by jodh-intel**
The Kata project doesn't use `OWNERS` files so rework the "Project
Maintainers" section of the contribution guide to explain who the
Approvers are and how the approval process works.

Fixes #30.

Signed-off-by: James O. D. Hunt <james.o.hunt@intel.com>

### CODE REVIEW

**Comment by jodh-intel** [comment]
@grahamwhaley, @egernst, @klynnrif - ptal.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -223,20 +223,26 @@ channels, Slack channels, and mailing lists:
...
 * [Mailing lists](http://lists.katacontainers.io/).
 
-### Project maintainers
+## Project maintainers
 
 The Kata Containers project maintainers are the people accepting or
-rejecting any PR. They are listed in the `OWNERS` files. There can be one
-`OWNERS` file per directory.
+rejecting any PR. Although [anyone can review PRs](#reviews), only the
+acknowledgement (or "ack") from an Approver counts towards a PR becoming approved.
```
Comment (klynnrif) [inline CONTRIBUTING.md]: ...counts towards the approval of a PR.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -223,20 +223,26 @@ channels, Slack channels, and mailing lists:
...
 
-The `OWNERS` files split maintainership into 2 categories: reviewers and
-approvers. All approvers also belong to the reviewers list. There must be at
-least one approval from one member of each list for a PR to be merged.
+Approvers are listed in github teams, one for each repository. The project
```
Comment (klynnrif) [inline CONTRIBUTING.md]: Approvers are listed in GitHub teams, one for each repository.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -223,20 +223,26 @@ channels, Slack channels, and mailing lists:
...
+uses the [Pull Approve](https://pullapprove.com) service meaning each
+repository contains a top level `.pullapprove.yml` configuration file. This
+file lists the github team used for that repositories approvals.
```
Comment (klynnrif) [inline CONTRIBUTING.md]: file lists the GitHub team responsible for that repository's approvals.

**Comment by jodh-intel** [comment]
Thanks @klynnrif - branch updated.

**Comment by jodh-intel** [comment]
Another doc PR that has been lingering for a while.

This is a quick one to review as I've basically just removed some outdated information.

**Comment by WeiZhang555** [comment]
LGTM

**Pull request merged**